src/librustc_codegen_llvm/intrinsic.rs

```diff
@@ -1251,7 +1252,7 @@ fn generic_simd_intrinsic(
         // trailing bits.
         let expected_int_bits = in_len.max(8);
         match ret_ty.kind {
-           ty::Uint(i) if i.bit_width() == Some(expected_int_bits) => (),
+           ty::Uint(i) if i.bit_width() == Some(expected_int_bits as usize) => (),
```

Comment:
Ugh, pretty much any use of `bit_width` is wrong, since it's sort of a "syntactic" measure.
Also, they don't work with `usize` so they break in potentially useful situations.

Reply:
Fair, but... this is old code and also what should it do instead?

Reply:
Ideally get the integer bitwidth from the layout? Most of the time you don't even need to call `layout_of`, since the layout is already present in some other variable.

Reply:
Is there a utility function for that?

Reply:
There's a `.bits()` method on `Size`.

Reply:
Sure but I don't have a `TyLayout`. All we have is a `Ty`.

Reply:
This file is a bit ridiculous. `fn_abi` has all the layouts for the signature inputs and output, but then they're also extracted separately for no good reason whatsoever.

`ret_ty` is always going to be equal to `fn_abi.ret.layout.ty`, and it really shouldn't exist by itself IMO. If only I hadn't sniped myself into a handful of other refactors at the same time, I would probably have a go at it.

Reply:
Can we agree to leaving fixing that mess to a future PR?

Reply:
Sound reasonable, sorry for dragging this on for so long.